app/src/main/java/com/twilio/video/app/participant/ParticipantViewState.kt

```diff
@@ -0,0 +1,23 @@
+package com.twilio.video.app.participant
+
+import com.twilio.video.RemoteParticipant
+import com.twilio.video.VideoTrack
+
+data class ParticipantViewState(
+    val sid: String? = null,
+    val identity: String? = null,
+    val videoTrack: VideoTrack? = null,
+    val muted: Boolean = false,
+    val mirror: Boolean = false,
+    val isPinned: Boolean = false,
+    val isScreenSharing: Boolean = false,
+    val isDominantSpeaker: Boolean = false,
+    val isLocalParticipant: Boolean = false
+)
+
+fun buildParticipantViewState(remoteParticipant: RemoteParticipant) =
+    ParticipantViewState(
+            remoteParticipant.sid,
+            remoteParticipant.identity,
+            remoteParticipant.remoteVideoTracks.firstOrNull()?.remoteVideoTrack
```

Comment:
How does this work if a participant is sharing both a camera video track and a screen share track?

Reply:
Haven't quite gotten there yet, but I am planning to cache all the participant view states in a the `mutableParticipants` property. Then, when we call the `participantThumbnails` getter function it will run an algorithm to determine what participants should be within the thumbnails and return that list. Similarly when calling the `primaryParticipant`, it will determine what video track should be rendered based on a similar algorithm. 

For the primary view algorithm I'm planning to follow what @timmydoza did with primary screen prioritization which is:
```
return pinnedParticipant || screenShareParticipant || dominantSpeaker || participants[0] || localParticipant;
```